libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -628,7 +628,10 @@ public String getNewAuthToken() {
                     Log.w("AccMgrAuthTokenProvider:fetchNewAuthToken", "accountManager.getAuthToken returned null bundle");
                 } else {
                     newAuthToken = bundle.getString(AccountManager.KEY_AUTHTOKEN);
-                    newInstanceUrl = bundle.getString(AuthenticatorService.KEY_INSTANCE_URL);
+                    final String encryptedInstanceUrl = bundle.getString(AuthenticatorService.KEY_INSTANCE_URL);
```

Comment:
We send the ```instanceUrl``` encrypted, but never bothered to decrypt it. I guess it was working because we never did any org split testing where the ```instanceUrl``` actually changes.